### DISCUSSION TITLE
Problemi dopo aggiornamento HA 2022.12

### DISCUSSION THREAD

**Discussion by baldus88** [indomus/forum#2171] [HUB: Home Assistant]
Ciao.
Mi avvalgo ancora della vostra disponibilità per aiutarmi ad uscire ad altri problemi che mi porto dietro dall'aggiornamento che avevo citato anche in un precedente post.

In primis nella homepage di homeassistant non ci sono più le voci di logbook, history etc

![image](https://user-images.githubusercontent.com/57867040/207147359-5f5c77c5-edcf-4e2a-acba-4cca169ab6ca.png)

Andando a spulciare i log di start vedo questo:

```
2022-12-12 19:49:09.913 ERROR (MainThread) [homeassistant.setup] Error during setup of component recorder Traceback (most recent call last):
2022-12-12 19:49:10.306 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of history_stats. Setup failed for dependencies: recorder
2022-12-12 19:49:10.311 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform history_stats.sensor:(DependencyError(...), 'Could not setup dependencies: recorder')
```

Avevo letto in giro per i forum che potrebbe essere dovuto al database delle statistiche che è corrotto, l'ho eliminato e ho forzato la ricreazione ma è rimasto tutto come vedete.

In aggiunta tra gli altri errori nei log vedo pure questo:

```
2022-12-12 19:48:58.773 ERROR (MainThread) [homeassistant.components.mqtt.mixins] Manually configured MQTT switch(s) found under platform key 'switch', please move to the mqtt integration key, see https://www.home-assistant.io/integrations/switch.mqtt/#new_format

2022-12-12 19:49:09.636 ERROR (MainThread) [homeassistant.components.mqtt.mixins] Manually configured MQTT sensor(s) found under platform key 'sensor', please move to the mqtt integration key, see https://www.home-assistant.io/integrations/sensor.mqtt/#new_format

2022-12-12 19:48:58.842 ERROR (MainThread) [homeassistant.components.switch] The mqtt platform for the switch integration does not support platform setup. Please remove it from your config.
```

Il fatto che io la configurazione mqtt l'ho tutta portata alla nuova configurazione quindi non ci sono sezioni dove la platform è mqtt e che magari andrebbe rimossa come dice... non capisco dove può andarla a pescare.

Questo sono alcune estrazioni:

```yaml
mqtt:
  binary_sensor:
    - name: "Finestra xx"
      state_topic: "tele/RF_Bridge/RESULT"
      value_template: '{{value_json.RfReceived.Data}}'
      payload_on: "xx"
      payload_off: "xx"
      device_class: Door
      qos: 1
    - name: "Portone"
      state_topic: "tele/RF_Bridge/RESULT"
      value_template: '{{value_json.RfReceived.Data}}'
      payload_on: "xx"
      payload_off: "xx"
      device_class: Door
      qos: 1
    sensor:
    - name: "Data aggiornamento"
      state_topic: "tele/sonoff_pow/SENSOR"
      value_template: "{{value_json['ENERGY'].TotalStartTime }}"
      qos: 1
      unit_of_measurement : "Date"
    - name: "Assorbimento TOTALE"
      state_topic: "tele/sonoff_pow/SENSOR"
      value_template: "{{value_json['ENERGY'].Total }}"
      qos: 1
      unit_of_measurement : "kWh"
  cover: 
    - name: Tapparella xx
      command_topic: "shellies/shellyswitch25-xx/roller/0/command"
      position_topic: "shellies/shellyswitch25-xx/roller/0/pos"
      set_position_topic: "shellies/shellyswitch25-xx/roller/0/command/pos"
      availability_topic: "shellies/shellyswitch25-xx/online"
      payload_available: "true"
      payload_not_available: "false"
      qos: 1
      retain: false
      payload_open: "open"
      payload_close: "close"
      payload_stop: "stop"
      position_open: 100
      position_closed: 0
      optimistic: false
```

Mi potete dare una mano per capire dove sta l'inghippo?

Grazie ancora

**Comment by snakuzzo** [comment]
Innanzi tutto, se non specifichi che tipo di installazione hai, è impossibile darti una mano.
In base ai primi log che hai postato, probabile tu stia usando installazione in venv e potrebbe trattarsi di un problema di dipendenze a livello software (ma me ne vado per un'idea eh). Se questo fosse il caso, ti suggerisco seriamente di passare a docker (magari con copmose) per evitare problemi futuri simili.

Quanto agli altri errori:
1. Hai definito qualcosa tu (o qualche integrazione/custom component) relativa a questo `mixins` che non so cosa sia
2. Il problema relativo allo `switch` è relativo a qualche tua errata configurazione (indentazione?) da qualche parte
3. Per HACS...svuota la cache...puoi anche provando un refresh della pagina con `CTRL+R`

**Reply by baldus88** [reply]
Ciao.
Si scusami ho dimenticato sono su HA Core quindi venv. 
In passato ero passato da HASSIO a docker, ma era quel periodo che avevano pensato di dismetterlo e ho deciso di fare il salto ad ha core ma sicuramente bisogna stare dietro a più componenti software che non sono lampanti subito quando procedi ad aggiornare la release.

Per quanto riguarda "mixins" è la prima volta che lo sento, proverò a spulciare la configurazione compreso pure dello switch... 

HACS in realtà non mi da problemi manca tutto il resto delle statistiche, logbook etc. Infatti se provo ad aprire un sensore x non mi fa vedere il grafico:

![image](https://user-images.githubusercontent.com/57867040/207151368-e9661884-6f82-48a4-9259-98e444994cd8.png)

Penso che sia collegato al problema iniziale che cita il log su history_stats...

**Reply by snakuzzo** [reply]
Non "pensi"...è sicuramente così.
Io non ti sto dicendo di passare ad HA OS o HA Supervised...ma di passare a Core su Docker.
Facci un pensierino ;)

**Reply by baldus88** [reply]
Sisi ma infatti era una delle strade che volevo intraprendere, tra l'altro leggevo dalle vostre guide che è quello che caldamente consigliate rispetto allo stesso HASSIO (per scarse performance e meno possibilità di gestione del resto) e di venv che avete abbondato di documentare, giusto??

https://indomus.it/focus/home-assistant-hassio-o-home-assistant-core-quale-installazione-fa-al-caso-mio/

**Reply by snakuzzo** [reply]
Guarda, l'obiettivo di quell'articolo è quello di farti riflettere e capire quale tipo di installazione faccia al caso tuo...non devi chiedere a noi cosa ti conviene.
Hai scelto di utilizzare core per i tuoi motivi... io posso dirti solo che, se utilizzi core, ti conviene utilizzarlo containerizzato e non con installazione in venv.

**Reply by baldus88** [reply]
Sono d'accordo, era giusto una riflessione "a voce alta" :) Tanto penso l'esperienza sul campo e sul prodotto che hai utilizzato ti fa capire sempre meglio di più come muoverti.

**Reply by snakuzzo** [reply]
Certo.
Diciamo che se sei proprio a digiuno, meglio dirottare su HA OS
Se invece già sei un po' più pratico o comunque hai voglia di studiare, provare, imparare senza fretta...magari con il passaggio a Core e l'utilizzo di Docker hai anche l'occasione di conoscere cose nuove e che al 101% ti saranno utili

**Comment by baldus88** [comment]
Ok ottimo sono riuscito a scovare quel famoso mqtt.mixins che rompeva le scatole... era richiamato nella configuration da un altro yaml collegato a zigbee2mqtt che avevo fatto mesi e mesi fa e non ricordavo più. Infatti non mi tornava di avere uno switch di tipo mqtt ma tantè ora non dovrebbe più rompere che l'ho commentato.

Per quanto riguarda il menu a sinistra l'ho ripristinato dopo avere rigenerato il db inserendo la voce "recorder" configuration file.

Grazie a tutti
